`azurerm_sql_server` - support for the `extended_auditing_policy` property (#5036)
terraform-providers/terraform-provider-azurerm: website/docs/r/sql_server.html.markdown

  location = "West US"
}

resource "azurerm_storage_account" "example" {

>   name                     = "accstr"

Reviewer: could this be 
```suggestion
  name                     = "examplesa"
```

* `identity` - (Optional) An `identity` block as defined below.

> * `blob_extended_auditing_policy` - (Optional) A `blob_extended_auditing_policy` block as defined below.

Reviewer: ```suggestion
* `extended_auditing_policy` - (Optional) A `blob_extended_auditing_policy` block as defined below.
```

-> You can access the Principal ID via `${azurerm_sql_server.example.identity.0.principal_id}` and the Tenant ID via `${azurerm_sql_server.example.identity.0.tenant_id}`

---

A `blob_extended_auditing_policy` block supports the following:

> * `storage_account_access_key` - (Required) Specifies the identifier key of the auditing storage account.

Reviewer: I think this should be
```suggestion
* `storage_account_access_key` - (Required) Specifies the access key to use for the auditing storage account.
```

* `storage_endpoint` - (Required) Specifies the blob storage endpoint (e.g. https://MyAccount.blob.core.windows.net).

> * `is_storage_secondary_key_in_use` - (Optional) Specifies whether storageAccountAccessKey value is the storage's secondary key.

Reviewer: ```suggestion
* `is_storage_secondary_key_in_use` - (Optional) Specifies whether `storage_account_access_key` value is the storage's secondary key.
```

if this is the case, would it make sense to add a new `storage_account_secondary_access_key` that conflicts with `storage_account_access_key` and sets `is_storage_secondary_key_in_use` to true?

> * `storage_secondary_key_enabled` - (Optional) Specifies whether `storage_account_access_key` value is the storage's secondary key.

Reviewer: ```suggestion
* `storage_account_access_key_is_secondary` - (Optional) Specifies whether `storage_account_access_key` value is the storage's secondary key.
```

> * `storage_account_access_key_is_secondary` - (Optional) Specifies whether `storage_account_access_key` value is the storage's secondary key.

Reviewer: could we order these alphabetically?

Author: So we don't need to put `required` fields in advance? We just order all `required` and `optional` fields alphabetically, right?

Reviewer: WHoops missed the optional, yes typically alphabetically, but sometimes i'll order for aesthetics ie `storage_endpoint` -> `storage_account_access_key` -> `storage_account_access_key_is_secondary`

> * `retention_days` - (Optional) Specifies the number of days to keep in the audit logs in the storage account.

Reviewer: ```suggestion
* `retention_days` - (Optional) Specifies the number of days to retain logs for in the storage account.
```
